selector/diversity.py

```diff
@@ -375,15 +375,25 @@ def hypersphere_overlap_of_subset(x: np.ndarray, x_subset: np.array) -> float:
     Agrafiotis, D. K.. (1997) Stochastic Algorithms for Maximizing Molecular Diversity.
     Journal of Chemical Information and Computer Sciences 37, 841-851.
     """
-    d = len(x_subset[0])
-    k = len(x_subset[:, 0])
+
     # Find the maximum and minimum over each feature across all molecules.
     max_x = np.max(x, axis=0)
     min_x = np.min(x, axis=0)
-    # normalization of each feature to [0, 1]
-    if np.any(np.abs(max_x - min_x) < 1e-30):
-        raise ValueError("One of the features is redundant and causes normalization to fail.")
 
+    if np.all(np.abs(max_x - min_x) < 1e-30):
+        raise ValueError("All of the features are redundant which causes normalization to fail.")
+
+    # Remove redundant features
+    non_red_feat = np.abs(max_x - min_x) > 1e-30
+    x = x[:, non_red_feat]
+    x_subset = x_subset[:, non_red_feat]
+    max_x = max_x[non_red_feat]
+    min_x = min_x[non_red_feat]
+
+    d = len(x_subset[0])
+    k = len(x_subset[:, 0])
+
```

Comment:
Shall we add a warning here to show the user that we have done the truncation?

Reply:
I am not sure, initially, I did not add a warning because the removal of columns with all elements equal should not affect the impact of each element on the diversity of the set, and therefore should not affect the selection. I don't know if it should be taken into account for other considerations.

Reply:
This is a really minor issue. I am going to close it. Thanks for the clarification.